### DISCUSSION TITLE
Hard-coded parameter is not rendered

### DISCUSSION THREAD

**Discussion by hippietrail** [NationalSecurityAgency/ghidra#7003] [Q&A]
I've got some instructions where the single operand is a 1-byte offset from a known address, `0x8300` in this case. In this form they work fine.
```
:DEC @>base8300    is op=0x92 ;base8300 { }
```
But I've also tried supporting specific outputs when the operand has a certain value. In this case the operand `0x75`, which would be equivalent to the address `0x8375` should render as `@key`. But the operand is not rendered, just the instruction.
```
:DEC @"key"    is op=0x92 ;p1s=0x75 { }
```
This is to mimic the output of my reference disassembler for this arch as seen in the comment column at the right. I do realize the 'proper' way to do this is at a different level where I just render it as an address and elsewhere I have symbols with the same values as those addresses. But I don't know how to do that in Ghidra yet, and my feeling is that things like this should also work.

No error results and the instruction pointer doesn't get out of sync so it's working except for the rendering.
<img width="766" alt="image" src="https://github.com/user-attachments/assets/1e425a94-873e-4cbb-8061-4911d5ba1dec">

Note also that the next instruction, which has two operands does render! Here's the code for that:
```
:SLL >p2s, @"key"   is op=0xE2 ;p1s=0x75 ;p2s { }
```

**Comment by GhidorahRex** [comment]
I would put the `@` sign inside the quote marks as well, but I can't see why that would be the issue either. Then try the debug sleigh instruction script I mentioned in the other question. Is it possible that you have another variant of `dec` that has no operands that is matching it first?

Another useful script for sleigh development if you're not aware of it would be the `ReloadSleighLanguage` script. It allows you to make sleigh changes in your editor and reload the language without having to open/close the program you're working with.

**Reply by hippietrail** [reply]
Thanks. I was initially putting everything inside quote marks but when things got unwieldy I did some experiments and found that only letters seemed to need to be quoted. Is this covered in the docs somewhere?
So far I've been quitting Ghidra and starting it again from Eclipse for every change, which is not fast. There is a ton I don't know yet. I'll have a look at the `ReloadSleighLanguage` script. Thanks.

**Reply by GhidorahRex** [reply]
Nope, definitely not fast.

I don't think it should need to be quoted, and that isn't documented anywhere, but I'm trying to narrow down possible exceptions. You can also right-click on the `DEC` instruction in the listing window and look at which lines of the sleigh file are being used by the instruction.
